### PR TITLE
V22F-683 black box tests

### PR DESCRIPTION
**Pull request by KungFuDonkey**
This PR tests if the blackbox responds quickly enough and logs a warning if it is not quick enough

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp [hunk 1]
```diff
@@ -38,7 +41,18 @@ void Driver::InitTrack(tTrack* p_track, void* p_carHandle, void** p_carParmHandl
     m_humanDriver.init_track(m_index, p_track, p_carHandle, p_carParmHandle, p_situation);
     m_humanDriver.SetRecorder(useRecorder ? m_recorder : nullptr);
 
+    auto start = std::chrono::system_clock::now();
     SMediator::GetInstance()->RaceStart(p_track, p_carHandle, p_carParmHandle, p_situation, m_recorder);
+    auto end = std::chrono::system_clock::now();
+
+    long milliseconds = static_cast<long>(std::chrono::duration_cast<std::chrono::milliseconds>(end - start).count() / BLACK_BOX_TESTS);
```
Comment (noutheesink) [inline speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp:48]: Add comment explaining why you divide by BLACK_BOX_TESTS (I know why but future devs won't)

**Review by noutheesink — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CMakeLists.txt [hunk 2]
```diff
@@ -59,14 +59,14 @@ SET(_HEADERS
...
-ADD_SDLIB_INCLUDEDIR(tgf math portability frontend rppUtils)
+ADD_SDLIB_INCLUDEDIR(tgf math portability frontend rppUtils unitTestUtilities)
 
 SD_ADD_LIBRARY(backend STATIC ${_SOURCES} ${_HEADERS} ${_INLINES})
 SET_TARGET_PROPERTIES(backend PROPERTIES LINKER_LANGUAGE CXX)
 
 TARGET_LINK_LIBRARIES(backend frontend)
 
-ADD_SDLIB_LIBRARY(backend tgf portability)
+ADD_SDLIB_LIBRARY(backend tgf portability unitTestUtilities)
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CMakeLists.txt:69]: Using unitTestUtiltities in non unit tests, should move the utilities to RPPUtils or another file

File: speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp [hunk 3]
```diff
@@ -38,7 +41,18 @@ void Driver::InitTrack(tTrack* p_track, void* p_carHandle, void** p_carParmHandl
     m_humanDriver.init_track(m_index, p_track, p_carHandle, p_carParmHandle, p_situation);
     m_humanDriver.SetRecorder(useRecorder ? m_recorder : nullptr);
 
+    auto start = std::chrono::system_clock::now();
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp:44]: Why is this done in the driver and not in the mediator? Right now this means only the assisted driver tests this but for example the replay driver doesn't

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
There are 2 csv files in the data folder that should be deleted.

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CMakeLists.txt (hunk 2 above)
Comment (KungFuDonkey) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CMakeLists.txt:69]: not possible as they need to be build, I can only move them to backend if thats what you want

File: speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp (hunk 3 above)
Comment (KungFuDonkey) [reply speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp:44]: cuz I need to log which is kinda frontend and if you want to display a message later you would have to move it again anyway

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CMakeLists.txt (hunk 2 above)
Comment (KungFuDonkey) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/CMakeLists.txt:69]: But I am only using something from the unitTestUtilities as they allow me to create runtime unit tests for the bb, so its not even that weird to do this

**Review by noutheesink — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (noutheesink) [reply speed-dreams/source-2.2.3/src/drivers/assistedhuman/Driver.cpp:48]: Committing this suggestion with @KungFuDonkey approval

**Review by noutheesink — APPROVED** [review]

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Move the test to the ResearchMenu when a blackbox is selected with a label like with database connecting. (Something like `Testing...` and `Failed: [error message]` or `Connection Succesfull`)

A log message is hard to see and requires launching the program through command line

**Review by noutheesink — APPROVED** [review]

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
image

**Review by suppergerrie2 — APPROVED** [review]

**Pull request merged**